servers/rendering/renderer_rd/renderer_scene_render_rd.cpp

```diff
@@ -585,6 +592,13 @@ void RendererSceneRenderRD::_render_buffers_post_process_and_tonemap(const Rende
 			fsr->fsr_upscale(rb, source_texture, dest_texture);
 		}
 
+		if (dest_is_msaa_2d) {
+			// We can't upscale directly into our MSAA buffer so we need to do a copy
```

Comment:
Why can't we upscale directly into an MSAA buffer?

Reply:
It wasn't working for me, I think it's because this is a compute shader and you need a different code path/can't write into MSAA textures.